File: DotNet/CesiumLanguageWriter/Advanced/ICesiumReferenceValuePropertyWriter.cs
namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// A writer that can write a value as a reference to another property.
    /// </summary>
    public interface ICesiumReferenceValuePropertyWriter : ICesiumPropertyWriter
    {
        /// <summary>
        /// Writes the value as a reference to another property.
        /// </summary>
        /// <param name="value">The reference.</param>
        void WriteReference(Reference value);

        /// <summary>
        /// Writes the value as a reference to another property.
        /// </summary>
        /// <param name="value">The reference.</param>
        void WriteReference(string value);

        /// <summary>
        /// Writes the value as a reference to another property.
        /// </summary>
        /// <param name="identifier">The identifier of the object which contains the referenced property.</param>
        /// <param name="propertyName">The property on the referenced object.</param>
        void WriteReference(string identifier, string propertyName);

        /// <summary>
        /// Writes the value as a reference to another property.
        /// </summary>
        /// <param name="identifier">The identifier of the object which contains the referenced property.</param>
        /// <param name="propertyNames">The hierarchy of properties to be indexed on the referenced object.</param>
        void WriteReference(string identifier, string[] propertyNames);
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/ICesiumReferenceValuePropertyWriter.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;

/**
 * A writer that can write a value as a reference to another property.
 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public interface ICesiumReferenceValuePropertyWriter extends ICesiumPropertyWriter {
    /**
    * Writes the value as a reference to another property.
    * @param value The reference.
    */
    void writeReference(Reference value);

    /**
    * Writes the value as a reference to another property.
    * @param value The reference.
    */
    void writeReference(String value);

    /**
    * Writes the value as a reference to another property.
    * @param identifier The identifier of the object which contains the referenced property.
    * @param propertyName The property on the referenced object.
    */
    void writeReference(String identifier, String propertyName);

    /**
    * Writes the value as a reference to another property.
    * @param identifier The identifier of the object which contains the referenced property.
    * @param propertyNames The hierarchy of properties to be indexed on the referenced object.
    */
    void writeReference(String identifier, String[] propertyNames);
}